crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+/// To check the final position of a UI element, read its [`Transform](bevy_transform::Transform) component.
+#[derive(
+    Component,
+    Deref,
+    DerefMut,
+    Copy,
+    Clone,
+    PartialEq,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Offset(pub UiRect<Val>);
+
+/// Controls the size of UI nodes
+///
+/// Layout is performed according to the [`LayoutStrategy`]
+/// To check the actual size of a UI element, read its [`Transform](bevy_transform::Transform) component
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct SizeConstraints {
+    /// The minimum extent, which cannot be violated by the layouting algorithm
+    pub min: Size<Val>,
+    /// The suggested extent, which will be used if other constraints can be comfortably satisfied
+    pub suggested: Size<Val>,
+    /// The maximum extent, which cannot be violated by the layouting algorithm
+    pub max: Size<Val>,
+    /// The expected aspect ratio, computed as width / height
+    pub aspect_ratio: Option<f32>,
+}
+
+impl SizeConstraints {
+    ///```rust
+    /// assert_eq!(SizeConstraints::DEFAULT, SizeConstraints::default())
+    ///```
+    pub const DEFAULT: SizeConstraints = SizeConstraints {
+        min: Size::DEFAULT,
+        suggested: Size::DEFAULT,
+        max: Size::DEFAULT,
+        aspect_ratio: None,
+    };
+
+    pub const FULL: SizeConstraints = SizeConstraints {
+        min: Size::DEFAULT,
+        suggested: Size::FULL,
+        max: Size::FULL,
+        aspect_ratio: None,
+    };
+
+    /// Sets only the minimum extent
+    pub const fn min(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            min: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the suggested extent
+    pub const fn suggested(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            suggested: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+
+    /// Sets only the suggested extent
+    pub const fn max(width: Val, height: Val) -> SizeConstraints {
+        SizeConstraints {
+            max: Size::new(width, height),
+            ..Self::DEFAULT
+        }
+    }
+}
+
+/// The space around and inside of a UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Spacing {
+    /// The space around the outside of the UI element
+    pub margin: UiRect<Val>,
+    /// The space around the inside of the UI element
+    pub padding: UiRect<Val>,
+    /// The space around the outside of the UI element that can be colored to create a visible border
+    pub border: UiRect<Val>,
```

Comment:
How exactly can the border be colored? We should link to the related component here

Reply:
It can't be (yet) 🥲